src/pages/accommodations/AccommodationForm.tsx

```diff
@@ -0,0 +1,307 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Box, Button, Stack, TextField, Typography } from '@mui/material';
+import { DateField, LocalizationProvider } from '@mui/x-date-pickers';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import dayjs from 'dayjs';
+import { Controller, useForm } from 'react-hook-form';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { useCreateAccommodation } from '@/api/mutations/accommodations/useCreateAccommodation';
+import { useDeleteAccommodation } from '@/api/mutations/accommodations/useDeleteAccommodation';
+import { useUpdateAccommodation } from '@/api/mutations/accommodations/useUpdateAccommodation';
+import { useGetAccommodation } from '@/api/queries/accommodations/useGetAccommodation';
+import YandexMap from '@/components/YandexMap';
+import { ROUTES } from '@/config/routes.config';
+import { AccommodationReq, accommodationSchema } from '@/types/accommodation.types';
+
+export default function AccommodationForm() {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const { data: accommodation } = useGetAccommodation({ id });
+  const { mutate: createAccommodation } = useCreateAccommodation();
+  const { mutate: updateAccommodation } = useUpdateAccommodation();
+  const { mutate: deleteAccommodation } = useDeleteAccommodation();
+
+  const {
+    watch,
+    control,
+    setValue,
+    handleSubmit,
+    formState: { errors, dirtyFields },
+  } = useForm<AccommodationReq>({
+    resolver: zodResolver(accommodationSchema),
+    defaultValues: {
+      title: '',
+      description: '',
+      allowedNumberOfPeople: 1,
+      availableFrom: String(new Date()),
+      availableTo: String(new Date(dayjs().add(1, 'week').toDate())),
+      thumbnailUrl: '',
+      numberOfRooms: 1,
+      previewImgUrl: '',
+      price: 0,
+      squareMeters: 0,
+      address: {
+        street: '',
+        city: '',
+        country: '',
+        zipCode: '',
+        latitude: 0,
+        longitude: 0,
+      },
+    },
+    values: accommodation?.data,
+  });
+
+  const latitudeWatch = watch('address.latitude');
+  const longitudeWatch = watch('address.longitude');
+
+  const onSubmit = (data: AccommodationReq) => {
+    if (id) {
```

Comment:
please make it flat
if (!id) {
   createAccommodation
  return
}
if (!Object.keys(dirtyFields).length) {
...
return
}
updateAccommodation....